tests/conftest.py

```diff
@@ -46,17 +47,44 @@ def _run(code, **kwargs):
     yield _run
 
 
+def _request_token(request):
+    token = ""
+    token += request.module.__name__
+    token += ".%s" % request.cls.__name__ if request.cls else ""
+    token += ".%s" % request.node.name
+    return token
+
+
 @pytest.fixture(autouse=True)
 def snapshot(request):
     marks = [m for m in request.node.iter_markers(name="snapshot")]
     assert len(marks) < 2, "Multiple snapshot marks detected"
     if marks:
         snap = marks[0]
-        token = ""
-        token += request.module.__name__
-        token += ".%s" % request.cls.__name__ if request.cls else ""
-        token += ".%s" % request.node.name
-        with snapshot_context(token, *snap.args, **snap.kwargs) as snapshot:
+        token = _request_token(request)
+        with _snapshot_context(token, *snap.args, **snap.kwargs) as snapshot:
             yield snapshot
     else:
         yield
```

Comment:
why do we yield here? Is it safe to remove this else condition?

Reply:
this is a good question for @Kyle-Verhoog 

my best guess would be that we always need to yield for the pytest fixture, but this should be a generator no matter what 🤔

Reply:
> my best guess would be that we always need to yield for the pytest fixture

this